packages/ui/Button/Button.js

```diff
@@ -216,11 +216,17 @@ const ButtonBase = kind({
 				props.flip = iconFlip;
 			}
 
-			return (
-				<ComponentOverride {...props}>
-					{icon}
-				</ComponentOverride>
-			);
+			if (icon === true) {
+				return (
+					<ComponentOverride {...props} />
+				);
+			} else {
+				return (
+					<ComponentOverride {...props}>
+						{icon}
```

Comment:
How about
```suggestion
						{icon === true ? null : icon}
```
instead of `if/else`?